### PR TITLE
Fixing git viewing renamed revision

### PR DESCRIPTION
**Pull request by tulinkry**
I noticed that viewing revision of renamed files was broken. The getHistory expects a full path file while the find renamed name returns the name relative to the repository directory. This leads that for renamed files the revision was never found as the filename after renaming was invalid.

### CODE REVIEW

File: opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java [hunk 1]
```diff
@@ -321,7 +327,7 @@ protected String findOriginalName(String fullpath, String changeset)
             ABBREV_LOG,
             "--abbrev-commit",
             "--name-status",
-            "--pretty=format:commit %h%b",
+            "--pretty=format:commit %h%n%d",
```
Comment (tulinkry) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java]: %h - commit hash
%n - new line
%d - file references (renames, add, remove)

**Comment by vladak** [comment]
Would be nice if this could be covered by tests like Mercurial.

**Comment by tulinkry** [comment]
Renamed files are covered with tests, but I don't get why this wasn't caught.

**Comment by vladak** [comment]
It would be good to know. Otherwise the tests cannot be trusted.

**Comment by tulinkry** [comment]
The test are not actually testing the behaviour of findRenamedFile. Let me fix it.

**Comment by tulinkry** [comment]
Rewrote the tests so they're meaning full.

**Comment by tulinkry** [comment]
This effectivelly swallowed the exception so noone could really say what was wrong.
https://github.com/oracle/opengrok/blob/f98c36aeb22b23e626acb37c4c02db1c74b485d9/opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java#L218-L222

For this particular cases it mostly way IndexOutOBException for the substring operation.

**Comment by tulinkry** [comment]
I don't know why javadoc hates me suddenly :D

File: opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java [hunk 2]
```diff
@@ -215,10 +217,14 @@ private HistoryRevResult getHistoryRev(
              * not exist or internal git error occured.
              */
             result.success = (status == 0);
-        } catch (Exception exp) {
+        } catch (Exception exception) {
```
Comment (vladak) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java]: you mentioned that the out-of-bounds exception was swalloed by this catch. Is using `Exception` maybe too broad then ?

File: opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java (hunk 1 above)
Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java]: so this in fact does not change the output, or does it ?

File: opengrok-indexer/src/test/java/org/opengrok/indexer/history/GitRepositoryTest.java [hunk 3]
```diff
@@ -350,47 +337,44 @@ public void testGetHistoryForTwiceRenamed() throws Exception {
...
-                + "}\n";
+        runRenamedTest(Paths.get("moved", "renamed2.c").toString(), "67dfbe26", new_content);
+        runRenamedTest(Paths.get("moved", "renamed2.c").toString(), "1086eaf5", new_content);
+
+        runRenamedTest(Paths.get("moved", "renamed.c").toString(), "1086eaf5", new_content);
 
-        runRenamedTest(Paths.get("moved", "renamed.c").toString(), "b641394", exp_str);
-        runRenamedTest(Paths.get("renamed.c").toString(), "ce4c98e", exp_str);
+        // old content (before revision b6413947 inclusively)
+        runRenamedTest(Paths.get("moved2", "renamed2.c").toString(), "b6413947", old_content);
```
Comment (vladak) [inline opengrok-indexer/src/test/java/org/opengrok/indexer/history/GitRepositoryTest.java]: these look like a candidate for array + forEach() ?

File: opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java (hunk 2 above)
Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java]: or move the `filename` assignment outside of the `try` block ?

Comment (tulinkry) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java]: No to the problem was it wasn't even logged with the bad logger parameters.

Comment (tulinkry) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java]: I just changed it so it's being logged

File: opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java (hunk 1 above)
Comment (tulinkry) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java]: Not really.
%b is too verbose - body + files
%d is just the files

File: opengrok-indexer/src/test/java/org/opengrok/indexer/history/GitRepositoryTest.java (hunk 3 above)
Comment (tulinkry) [reply opengrok-indexer/src/test/java/org/opengrok/indexer/history/GitRepositoryTest.java]: Is it more readable though? I would leave it this way so a test failure can point to a particular line and also it's not that much lines.

File: opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java (hunk 2 above)
Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java]: Okay. Still large try/catch blocks with generic catch make me nervous.

File: opengrok-indexer/src/test/java/org/opengrok/indexer/history/GitRepositoryTest.java (hunk 3 above)
Comment (vladak) [reply opengrok-indexer/src/test/java/org/opengrok/indexer/history/GitRepositoryTest.java]: If you complement the assert with a message it should be enough.

File: opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java (hunk 1 above)
Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java]: I see.

File: opengrok-indexer/src/test/java/org/opengrok/indexer/history/GitRepositoryTest.java (hunk 3 above)
Comment (tulinkry) [reply opengrok-indexer/src/test/java/org/opengrok/indexer/history/GitRepositoryTest.java]: done

**Pull request merged**